content/knowledge/scalinghistory.md

```diff
@@ -0,0 +1,135 @@
+   Also the higher the adaptation, the higher the overall value represented by the overall native cryptocurrency and therefore higher the value that needs to be locked up as staking by the nodes.
+   This means the native currency's full potential isn't utilised as once they are locked, they can no longer be used for any other purposes unless they are unlocked after the locking period expires.
+   Network’s safety is proportional to the value of stake committed.
+
+## 3.1 Sharding and layer 2 solutions
+
+   For solving this scaling issue, 'sharding' is introduced(e.g. In case of [Ethereum2.0](https://www.youtube.com/watch?v=ctzGr58_jeI&t=657s&ab_channel=Finematics).
+   Sharding is a way of spreading the computing and storage workload from a blockchain network, so that each node no longer has to process the entire network's transactional load.
+   Each node only needs to maintain info corresponding to its specific shard or partition.
+   For example, addresses starting with “0x00” will be part of Shard1 and addresses starting with “0x01” will be part of Shard2 etc.
+   There is a mechanism in place for inter-shard communication so that the whole blockchain is still viewable from any node.
+   All the shards will be connected to the “Beacon chain” which will have all the history with it and acts as an orchestrator of the whole network.
+   The consensus between all the shards will be maintained by the Beacon chain.
+   However sharding has its own challenges.
+   Corrupting nodes in a given shard may lead to permanent loss of data.
+   One way to tackle this issue is by randomly assigning a node to a shard and randomly reassigning the node to another shard.
+   Sharding has another problem that the composability aspect of Smart contract compatible blockchain will no longer exist.
+   For example, if a Defi built on top of Ethereum1.0 ecosystem relies on or combines few other projects, then in a sharded system of Ethereum2.0, ideally they should be on the same shard which may not always be the case.
+   If they are not on the same shard then they all needs to be ok with asynchronous updates which is [not ideal](https://www.coindesk.com/tech/2020/10/13/will-a-sharded-ethereum-be-flexible-enough-for-decentralized-finance/).
+
+## 4. Avalanche
+
+   The next level solution for scalability in terms of consensus algorithms is by Avalanche network.
+   In Avalanche, the underlying [sybil control mechanism](https://en.wikipedia.org/wiki/Sybil_attack) is still PoS but there is a different mechanism in place to finalise validating transactions.
+   This is achieved by a particular node randomly subsampling from other nodes multiple times across multiple cycles and eventually arriving at a consensus.
+   This is then recorded in a Directed Acyclic Graph as opposed to a chain of Blocks.
+   This consensus can easily scale at a much higher level compared to Bitcoin/PoW or Ethereum2.0/PoS because of faster agreement of transactions.
```

Comment:
Scale _to_ a higher level. Because the communicational complexity is much lower than BFT; we only need to communicate with random sub-samples of the network for each round. If these are logarithmically smaller than the entire set of validators on the network, the performance doesn't decrease exponentially with the size of the validator set.